### PR TITLE
Hugo vite hmr

### PR DESCRIPTION
**Pull request by dawidprosba**
Changes:
- updated eslint config and converted it to .json as it's easier to read.
- added vite.js (first hugo is built without assets and then vite on top of hugo build and transpiles the assets)
- added hot module reloading with preact

### CODE REVIEW

File: exampleSite/package.json [hunk 1]
```diff
@@ -50,12 +40,14 @@
...
     "eslint-plugin-react": "^7.29.4",
     "eslint-plugin-react-hooks": "^4.3.0",
+    "fast-glob": "^3.2.11",
     "hugo-bin-extended": "^0.93.3",
     "inquirer": "^8.2.1",
     "jest": "^27.5.1",
     "npm-run-all": "^4.1.5",
     "onchange": "^7.1.0",
     "path": "^0.12.7",
+    "toml": "^3.0.0",
```
Comment (Lowess) [inline exampleSite/package.json:50]: `vite` should be part of devDeps I think

Comment (Lowess) [inline exampleSite/package.json:50]: `vite` should be part of devDeps I think

**Review by Lowess — CHANGES_REQUESTED** [review]

File: exampleSite/package.json (hunk 1 above)
Comment (dawidprosba) [reply exampleSite/package.json:50]: done

**Comment by dawidprosba** [comment]
Good catch, added the vite to dev deps

**Review by Lowess — APPROVED** [review]

**Pull request merged**